apps/site/assets/ts/stop/components/DepartureTimes.tsx

```diff
@@ -169,23 +174,47 @@ const DepartureTimes = ({
   departuresForDirection,
   onClick,
   alertsForDirection,
+  stopName: stopId,
   overrideDate
 }: DepartureTimesProps): ReactElement<HTMLElement> => {
   const groupedDepartures = groupBy(departuresForDirection, "trip.headsign");
   return (
     <>
-      {Object.entries(groupedDepartures).map(([headsign, departures]) => {
-        return (
-          <div
-            key={`${headsign}-${route.id}`}
-            onClick={() => onClick(route, directionId)}
-            onKeyDown={() => onClick(route, directionId)}
-            role="presentation"
-          >
-            {getRow(headsign, departures, alertsForDirection, overrideDate)}
-          </div>
-        );
-      })}
+      {Object.keys(groupedDepartures).length > 0 ? (
+        <>
+          {Object.entries(groupedDepartures).map(([headsign, departures]) => {
+            return (
+              <div
+                key={`${headsign}-${route.id}`}
+                onClick={() => onClick(route, directionId)}
+                onKeyDown={() => onClick(route, directionId)}
+                role="presentation"
+              >
+                {getRow(headsign, departures, alertsForDirection, overrideDate)}
+              </div>
+            );
+          })}
+        </>
+      ) : (
+        <div
+          key={`${route.direction_destinations[directionId]}-${route.id}`}
+          onClick={() => onClick(route, directionId)}
+          onKeyDown={() => onClick(route, directionId)}
+          role="presentation"
+        >
```

Comment:
opportunity: since they all do the same thing onClick, making a reusable component for clickable rows would reduce the need to retype this div and these attributes

Reply:
made a component but it looks pretty similar to typing this div 😅 
```
<DepartureRow
              key={`${headsign}-${route.id}`}
              route={route}
              directionId={directionId}
              onClick={() => onClick(route, directionId)}
              >
                {getRow(headsign, departures, alertsForDirection, overrideDate)}
              </DepartureRow>
       ```